wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -501,11 +501,16 @@ public void cancelAll() {
    * scheduled by the scheduler; it will not work on commands inside compositions, as the scheduler
    * does not see them.
    *
-   * @param commands the command to query
-   * @return whether the command is currently scheduled
+   * @param command a single command to check
+   * @param commands multiple commands to check
+   * @return whether all of the commands are currently scheduled
    */
-  public boolean isScheduled(Command... commands) {
-    return m_scheduledCommands.containsAll(Set.of(commands));
+  public boolean isScheduled(Command command, Command... commands) {
+    if (commands.length == 0) {
+      return m_scheduledCommands.contains(command);
+    } else {
+      return m_scheduledCommands.contains(command) && m_scheduledCommands.containsAll(Set.of(commands));
+    }
```

Comment:
A simple loop over the array would be fine

```java
public boolean isScheduled(Command... commands) {
  for (var command : commands) {
    if (!m_scheduledCommands.contains(command)) {
      return false;
    }
  }

  return true;
}  
```

Wrapping in `Set.of` is useful only if the same command object is passed in more than once, which should be rare or even nonexistent.

Note that a variadic argument will _always_ create a new array, even if no objects are passed in; ie, `isScheduled()` effectively compiles to `isScheduled(new Command[0])`, so this approach doesn't prevent array allocations. If you want to avoid array object allocations, overloads for some number of predetermined parameter counts would be necessary.

Reply:
That loop would make it match the C++ version too.

Reply:
Is it worth doing the predetermined overloads? multi-command schedule checks are already uncommon

Reply:
A single-command overload would probably be fine with varargs as a fallback